modular_skyrat/modules/ashwalkers/code/buildings/ash_farming.dm

```diff
@@ -1,42 +1,46 @@
 /datum/component/simple_farm
+	///whether you can actually farm it at the moment
+	var/allow_plant = FALSE
 	///whether we limit the amount of plants you can have per turf
 	var/one_per_turf = TRUE
 	///the reference to the movable parent the component is attached to
 	var/atom/atom_parent
 	///the amount of pixels shifted (x,y)
 	var/list/pixel_shift = 0
-
 /datum/component/simple_farm/Initialize(set_plant = FALSE, set_turf_limit = TRUE, list/set_shift = list(0, 0))
 	//we really need to check if its movable
 	if(!isatom(parent))
 		return COMPONENT_INCOMPATIBLE
 	atom_parent = parent
+	//important to allow people to just straight up set allowing to plant
+	allow_plant = set_plant
 	one_per_turf = set_turf_limit
 	pixel_shift = set_shift
 	//now lets register the signals
 	RegisterSignal(atom_parent, COMSIG_PARENT_ATTACKBY, PROC_REF(check_attack))
 	RegisterSignal(atom_parent, COMSIG_PARENT_EXAMINE, PROC_REF(check_examine))
-
 /datum/component/simple_farm/Destroy(force, silent)
 	//lets not hard del
 	UnregisterSignal(atom_parent, list(COMSIG_PARENT_ATTACKBY, COMSIG_PARENT_EXAMINE))
 	atom_parent = null
 	return ..()
-
 /**
  * check_attack is meant to listen for the comsig_parent_attackby signal, where it essentially functions like the attackby proc
  */
 /datum/component/simple_farm/proc/check_attack(datum/source, obj/item/attacking_item, mob/user)
 	SIGNAL_HANDLER
+	//if it behaves like a shovel
+	if(attacking_item.tool_behaviour == TOOL_SHOVEL)
```

Comment:
That part literally does nothing, that's why I removed it.

Reply:
Yeah and I restored it because this was a hacky fix to restore ashland farming, want me to get rid of it?

Reply:
> Yeah and I restored it because this was a hacky fix to restore ashland farming, want me to get rid of it?

You only need the three lines at the very bottom of the file that add the farming functionality to lavaland basalt.